src/integer/mat_z/inverse.rs

```diff
@@ -0,0 +1,134 @@
+// Copyright © 2023 Marcel Luca Schmidt
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! This module contains the implementation of the `inverse` function.
+
+use flint_sys::{
+    fmpz::fmpz,
+    fmpz_mat::{fmpz_mat_det, fmpz_mat_inv},
+};
+
+use crate::{
+    error::MathError,
+    integer::Z,
+    traits::{GetNumColumns, GetNumRows},
+};
+
+use super::MatZ;
+
+impl MatZ {
+    /// Returns the inverse of the matrix if it exists (is squared and
+    /// has a determinant unequal to zero) and has only [`Z`] values
+    /// (has a determinant of '1' or '-1').
+    ///
+    /// # Example
+    /// ```rust
+    /// use qfall_math::integer::MatZ;
+    /// use crate::qfall_math::traits::*;
+    ///
+    /// let mut matrix = MatZ::new(2, 2).unwrap();
+    /// matrix.set_entry(0, 0, 1);
+    /// matrix.set_entry(1, 1, 1);
+    /// let matrix2 = matrix.invert().unwrap();
```

Comment:
I feel like this example does not really provide any information to a user, please consider a different example